bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/MailingListArchiveReaderBot.java

```diff
@@ -114,7 +114,8 @@ synchronized void inspect(Conversation conversation) {
             pr = foundPr.get();
             resolvedPullRequests.put(first.id(), pr);
         }
-        var bridgeIdPattern = Pattern.compile("^[^.]+\\.[^.]+@" + pr.repository().url().getHost() + "$");
+        var bridgeIdStr = "^[^.]+\\.[^.]+@(?:" + pr.repository().url().getHost() + "|" + pr.repository().authenticatedUrl().getHost() + ")$";
+        var bridgeIdPattern = Pattern.compile(bridgeIdStr);
```

Comment:
I think the correct fix is to just get the host from `authenticatedUrl`. Perhaps there should be a different method for this kind of usecase that better describes what we need here (the raw actual host, without the pattern rewrite), but at least for now, I think changing this to `authenticatedUrl` is a good enough fix. Thanks for figuring this out.

Reply:
Just a reminder, If so, we should also change `return EmailAddress.from(encodedCommon + "." + UUID.randomUUID() + "@" + pr.repository().url().getHost())` in `ReviewArchive::getUniqueMessageId ` to `return EmailAddress.from(encodedCommon + "." + UUID.randomUUID() + "@" + pr.repository().authenticatedUrl().getHost())`

Reply:
Yes, right, we are currently generating `Message-ID`s with the rewritten url. That's bad.

Reply:
If we only filter the `authenticatedUrl`, what about the polluted data? The SKARA project now has many emails which have the message id with `git.openjdk.org`. If we don’t filter them, they would be posted as comments.

Reply:
> If we only filter the `authenticatedUrl`, what about the polluted data? The SKARA project now has many emails which have the message id with `git.openjdk.org`. If we don’t filter them, they would be posted as comments.

Only Skara and playground has been polluted. Maybe it's fine.

Reply:
Yes, but it's only the Skara project, so I was ok with living with it. But sure, we can keep the double filter if we add the following comment.
```suggestion
        // Filter on both original, and any pattern replaced, host for now as we have polluted
        // data in the Skara project mail archives. We should revert this to just authenticatedUrl
        // eventually.
        var bridgeIdStr = "^[^.]+\\.[^.]+@(?:" + pr.repository().url().getHost() + "|" + pr.repository().authenticatedUrl().getHost() + ")$";
        var bridgeIdPattern = Pattern.compile(bridgeIdStr);
```

Reply:
> Yes, but it's only the Skara project, so I was ok with living with it.

After thinking more, I am OK with it too. It can also verify and confirm my thought about the hosts in this patch.

Reply:
Another option for us is that we use double filter right now and after 14 days, we revert it back to only filter `authenticatedUrl()`. In this way, the polluted Skara emails wouldn't be bridged.

Reply:
Seems to only influence several PRs, so I don't want to complicate the flow.

Reply:
> Seems to only influence several PRs, so I don't want to complicate the flow.

Sure! I am also ok with it.